src/audio/multiband_drc/multiband_drc.c

```diff
@@ -0,0 +1,605 @@
+
+	comp_info(dev, "multiband_drc_free()");
+
+	comp_data_blob_handler_free(cd->model_handler);
+
+	rfree(cd);
+	rfree(dev);
+}
+
+static int multiband_drc_params(struct comp_dev *dev, struct sof_ipc_stream_params *params)
+{
+	int ret;
+
+	comp_info(dev, "multiband_drc_params()");
+
+	comp_dbg(dev, "multiband_drc_verify_params()");
+	ret = comp_verify_params(dev, 0, params);
+	if (ret < 0) {
+		comp_err(dev, "multiband_drc_params(): comp_verify_params() failed.");
+		return -EINVAL;
+	}
+
+	/* All configuration work is postponed to prepare(). */
+	return 0;
+}
+
+static int multiband_drc_cmd_get_data(struct comp_dev *dev,
+				      struct sof_ipc_ctrl_data *cdata,
+				      int max_size)
+{
+	struct multiband_drc_comp_data *cd = comp_get_drvdata(dev);
+	int ret = 0;
+
+	switch (cdata->cmd) {
+	case SOF_CTRL_CMD_BINARY:
+		comp_info(dev, "multiband_drc_cmd_get_data(), SOF_CTRL_CMD_BINARY");
+		ret = comp_data_blob_get_cmd(cd->model_handler, cdata, max_size);
+		break;
+	default:
+		comp_err(dev, "multiband_drc_cmd_get_data() error: invalid cdata->cmd");
+		ret = -EINVAL;
+		break;
+	}
+	return ret;
+}
+
+static int multiband_drc_cmd_set_data(struct comp_dev *dev,
+				      struct sof_ipc_ctrl_data *cdata)
+{
+	struct multiband_drc_comp_data *cd = comp_get_drvdata(dev);
+	int ret = 0;
+
+	switch (cdata->cmd) {
+	case SOF_CTRL_CMD_BINARY:
+		comp_info(dev, "multiband_drc_cmd_set_data(), SOF_CTRL_CMD_BINARY");
+		ret = comp_data_blob_set_cmd(cd->model_handler, cdata);
+		break;
+	default:
+		comp_err(dev, "multiband_drc_cmd_set_data() error: invalid cdata->cmd");
+		ret = -EINVAL;
+		break;
+	}
+
+	return ret;
+}
+
+static int multiband_drc_cmd(struct comp_dev *dev, int cmd, void *data, int max_data_size)
+{
+	struct sof_ipc_ctrl_data *cdata = data;
+	int ret = 0;
+
+	comp_info(dev, "multiband_drc_cmd()");
+
+	switch (cmd) {
+	case COMP_CMD_SET_DATA:
+		ret = multiband_drc_cmd_set_data(dev, cdata);
+		break;
+	case COMP_CMD_GET_DATA:
+		ret = multiband_drc_cmd_get_data(dev, cdata, max_data_size);
+		break;
+	default:
+		comp_err(dev, "multiband_drc_cmd(), invalid command");
+		ret = -EINVAL;
+	}
+
+	return ret;
+}
+
+static int multiband_drc_trigger(struct comp_dev *dev, int cmd)
+{
+	int ret;
+
+	comp_info(dev, "multiband_drc_trigger()");
+
+	ret = comp_set_state(dev, cmd);
+	if (ret == COMP_STATUS_STATE_ALREADY_SET)
+		ret = PPL_STATUS_PATH_STOP;
+
+	return ret;
+}
+
+static void multiband_drc_process(struct comp_dev *dev, struct comp_buffer *source,
+				  struct comp_buffer *sink, int frames,
+				  uint32_t source_bytes, uint32_t sink_bytes)
+{
+	struct multiband_drc_comp_data *cd = comp_get_drvdata(dev);
+
+	buffer_invalidate(source, source_bytes);
+
+	cd->multiband_drc_func(dev, &source->stream, &sink->stream, frames);
+
+	buffer_writeback(sink, sink_bytes);
+
+	/* calc new free and available */
+	comp_update_buffer_consume(source, source_bytes);
+	comp_update_buffer_produce(sink, sink_bytes);
```

Comment:
are we sure we want to produce on every drc process?

Reply:
I don't exactly know your meaning here. Aren't we?

Reply:
dont worry i misread the code, this is good, thought this was called per band